plugins/Login/Login.php

```diff
@@ -47,6 +48,15 @@ public function getStylesheetFiles(&$stylesheetFiles)
         $stylesheetFiles[] = "plugins/Login/stylesheets/variables.less";
     }
 
+    public function beforeSessionStart()
+    {
+        // if this is a login request & form_rememberme was set, change the session cookie expire time before starting the session
+        $rememberMe = Common::getRequestVar('form_rememberme', false);
```

Comment:
I'm not sure about this because all it takes be a URL parameter `?form_rememberme` at any point and it could potentially be misused. Can we set it as part of the regular form like it used to happen where we also check an nonce?

Reply:
`Session::rememberMe` has to be done before the session is started, or it will have no effect, and nonces are stored in the session so to read one we'd have to start a session. I'm not too sure how to make this better, but I'll think about it. If you have an idea would be interested.

Reply:
It would be always done before the session is started when setting the URL parameter in this case right? Ideally, it would be only set after a successful log in. Is there any way to move it kind of back to the original logic? I haven't thought of how it could work otherwise.

Reply:
It has to be set before the session is created, and before a login is attempted there is an active session (since the login form uses a nonce I think). It's super annoying, because Zend won't allow you to destroy & restart a session. Maybe there's some way to only execute this code if we're attempting a login, but I haven't thought of one yet.

Reply:
The "before session start" limitation is documented in the PHP function: https://secure.php.net/manual/en/function.session-set-cookie-params.php

Reply:
I guess we can check if the module = Login as well. If someone adds that to the URL, then they'll be taken to the login page. And I guess maybe check that if a session is already set the form_rememberme=1 param wouldn't change the expiry time.

Reply:
Doing it only when module=Login sounds good although it would maybe still have the affect that the session be active in the end for a longer time? Is it maybe possible to also only do it for `$_POST` requests? Would that help?

> It has to be set before the session is created
from what I saw it is needed to be called before session is started, not created but I may misunderstand it.

Reply:
Maybe we check for module=Login, action=login, and only in POST. If someone adds form_rememberme to this request, it's basically just ticking the checkbox.

> from what I saw it is needed to be called before session is started, not created but I may misunderstand it.

I don't think Zend differentiates between starting & creating a session. If there's no SESSIONID, it's created, otherwise just started. And there's always a session active even before login since the form uses a nonce (and Zend has a hardcoded exception that disallows creating a session after one is destroyed).